### ISSUE TITLE
NoOptionError when training model

### ISSUE THREAD

**Issue by christopherthomson** [inpho/topic-explorer#330]
I have installed topicexplorer on Ubuntu 16.04, created a directory with a few text files and run init and prep successfully. But I get an error about missing options when running 'topicexplorer train topicexplorer.ini'. The usage instructions don't seem to indicate other required arguments. I installed using Anaconda and Python 3.6 as per [the instructions here](https://github.com/inpho/topic-explorer#default-install-windows-users-should-use-developer-install-until-further-notice). Can anyone help?

``
Traceback (most recent call last):
  File "/home/chris/anaconda3/bin/topicexplorer", line 11, in <module>
    sys.exit(main())
  File "/home/chris/anaconda3/lib/python3.6/site-packages/topicexplorer/__main__.py", line 243, in main
    benchmark(train.main)(args)
  File "/home/chris/anaconda3/lib/python3.6/site-packages/topicexplorer/train.py", line 119, in main
    raise NoOptionError
TypeError: __init__() missing 2 required positional arguments: 'option' and 'section'
``

**Comment by christopherthomson** [comment]
I realised what this is: the -k argument is required. Can't do anything without number of topics! But the error message could be improved to indicate this.

**Comment by colinallen** [comment]
Actually, I'm reopening this issue as you should not have seen this error message -- here's what you should see when you call 'train' without specifying -k, i.e. a prompt for the number(s) of topics:
```
topicexplorer train <corpus.ini>
Number of Topics [Default '20 40 60 80']: 
```
What do you get for`topicexplorer version` ?

**Comment by christopherthomson** [comment]
1.0b216

Installed using pip.

**Comment by colinallen** [comment]
Could you please paste or attach the contents of your ini file?

**Comment by christopherthomson** [comment]
[DEFAULT]
htrc = False
htrc_metadata = None
sentences = False
certfile = None
keyfile = None
ca_certs = None
ssl = False
port = 8000
host = 127.0.0.1
icons = link
corpus_link = None
doc_title_format = {0}
doc_url_format =
raw_corpus = None
label_module = None
fulltext = False
pdf = False
topics =
cluster = None
corpus_desc = None
home_link = /
lang = None
tokenizer = default

[main]
path = /home/chris/Projects/models
corpus_file = /home/chris/Projects/models/topicexplorer-nltk-en-freq3-N129.npz
raw_corpus = /home/chris/Projects/topicexplorer/texts
sentences = False
corpus_desc = topicexplorer.ini.md
model_pattern = /home/chris/Projects/models/topicexplorer-nltk-en-freq3-N129-LDA-K{0}-document-270618-100.npz
context_type = document
topics = [5]

[www]
corpus_name = food
icons = fingerprint,link
fulltext = false
tokenizer = default

[logging]
path = logs/topicexplorer/{0}.log

**Comment by christopherthomson** [comment]
Sorry, you're right, I had re-run it at that point. I have created a new test instance, as follows:

$ topicexplorer init --name 'test2' ./texts/
Building corpus from ./texts/ with corpus_from_files function
100%|##################################################################################|
Saving corpus as /home/chris/Projects/test-te/models/texts.npz
Writing configuration file texts.ini

TIP: Only initalizing corpus object and config file.
     Next prepare the corpus using:
         topicexplorer prep texts.ini
     Or skip directly to training LDA models using:
         topicexplorer train texts.ini


Here is the config file generated:

[main]
path = /home/chris/Projects/test-te/models
corpus_file = /home/chris/Projects/test-te/models/texts.npz
raw_corpus = /home/chris/Projects/test-te/texts
sentences = False
corpus_desc = texts.ini.md

[www]
corpus_name = test2
icons = fingerprint,link
fulltext = false
tokenizer = default

[logging]
path = logs/texts/{0}.log

And here is the error message:

$ topicexplorer train texts.ini
Traceback (most recent call last):
  File "/home/chris/anaconda3/bin/topicexplorer", line 11, in <module>
    sys.exit(main())
  File "/home/chris/anaconda3/lib/python3.6/site-packages/topicexplorer/__main__.py", line 243, in main
    benchmark(train.main)(args)
  File "/home/chris/anaconda3/lib/python3.6/site-packages/topicexplorer/train.py", line 119, in main
    raise NoOptionError
TypeError: __init__() missing 2 required positional arguments: 'option' and 'section'

**Comment by colinallen** [comment]
Thanks - it looks to me as though the ini file is being incorrectly generated.  It should have the line

**Comment by colinallen** [comment]
Thanks - it looks to me as though the ini file is being incorrectly generated. It should have the line

```
topics =
```
in the [DEFAULT] section.

I will try to recreate the problem here.

**Comment by colinallen** [comment]
Still can't replicate the problem here but @JaimieMurdock is also looking at it now.

**Comment by JaimieMurdock** [comment]
Had fixed this in https://github.com/inpho/topic-explorer/commit/89513535cbde68bb6e9a623d1fa39265dd3a34ea but not yet released.

I've released 1.0b217 now which includes fixes for several other issues as well.
